File: easy/slang_flavor.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

func main() {
	var (
		phrases = []string{", yeah!", ", this is crazy, I tell ya.",
			", can U believe this?", ", eh?", ", aw yea.",
			", yo.", "? No way!", ". Awesome!"}
		c int
		l bool
		t byte = '\n'
	)
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	scanner := bufio.NewScanner(data)
	scanner.Split(bufio.ScanBytes)
	for scanner.Scan() {
		t = scanner.Text()[0]
		if t == '.' || t == '!' || t == '?' {
			l = !l
			if !l {
				fmt.Print(phrases[c])
				c = (c + 1) % len(phrases)
				continue
			}
		}
		fmt.Printf("%c", t)
	}
	if t != '\n' {
		fmt.Println()
	}
}

File: easy/slang_flavor.rb
phrases = [", yeah!",
	   ", this is crazy, I tell ya.",
	   ", can U believe this?",
	   ", eh?",
	   ", aw yea.",
	   ", yo.",
	   "? No way!",
	   ". Awesome!"]
i, l = 0, false
File.open(ARGV[0]).each_line do |line|
  s = line.split("").each { |x|
    if x == "." || x == "!" || x == "?"
      if l
        print phrases[i]
        i, l = (i+1) % 8, false
      else
        print x
        l = true
      end
    else
      print x
    end
  }
  puts if s.last != "\n"
end
